mvvm_cards/src/models/AppModel.ts

```diff
@@ -21,7 +22,18 @@ import { IAppModel } from "./i_appmodel";
 export class AppModel implements IAppModel {
     @observable
     _docTitle = "Main Document Title"
-
     get docTitle() { return this._docTitle; }
     set docTitle(title: string) { this._docTitle = title; }
+
+    @observable
+    _cards: ICard[] = []
+    get cards() { return this._cards; }
+    set cards(cards: ICard[]) { this._cards = cards; }
+
+     // in the current Mobx version we might not to write the action. Result will be the same
+    @action
+    addCard(card: ICard) { 
+        this._cards.unshift(card);
```

Comment:
it's because of our requirements. We need to add a new card at the top of a column

so let's think about a workflow
we'll have an array of cards and we'll iterate it in the View. So the item with the index 0 will be at the top (first item) and the item with the last index will be at the bottom.
So when we'll add a new item, the View will be re-rendered and the iteration cycle will happen again. In this case if we push our new item like the last item  ( and push adds the item ato the end of the array ) so in the View, it also will be rendered at the bottom of the column.

so that's why we need to at the new item to the beginning of our cards array (to render it at the top)

![image](https://user-images.githubusercontent.com/9897782/72661142-8e882a00-39df-11ea-8f08-b06f984251e4.png)